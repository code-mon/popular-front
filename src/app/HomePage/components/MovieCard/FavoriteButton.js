import React, { Component } from 'react';

//images
import star from '../../../assets/star.svg';
import starToggled from '../../../assets/star-toggled.svg';

//=========================
// COMPONENT
//=========================
const FavoriteButton = ( { isToggled } ) => {
    let tempStyle;
    isToggled ? tempStyle = { ...localStyles.favButton, ...localStyles.favToggled } : tempStyle = { ...localStyles.favButton, ...localStyles.favUntoggled };
    
    return (
        <div style={ tempStyle }></div>
    );
};

//=========================
// STYLes
//=========================
const localStyles = {
    favButton: {
        marginTop: 14,
        width: 30,
        height: 30,
        alignSelf: 'center',
        backgroundSize: 'contain',
    },

    favToggled: {
        backgroundImage: `url( ${ starToggled } )`,
    },

    favUntoggled:{
        backgroundImage: `url( ${ star } )`,
    }
}

//=========================
// EXPORTS
//=========================
export default FavoriteButton;