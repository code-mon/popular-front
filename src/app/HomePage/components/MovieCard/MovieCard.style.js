//=========================
// STYLES
//=========================
const styles = {
    base: {
        boxSizing: 'border-box',
        cursor: 'pointer',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        textAlign: 'center',
        width: '300px',
        height: '169px',
        fontSize: '18px',
        color: '#f5f5f5',
        opacity: '0.6',
        transition: 'transform 300ms ease-in',
        ':hover': {
            zIndex: '1',
            opacity: '1',
            fontSize: '22px',
            transform: 'scale(1.2)',
        }
    },
    hidden: {
        opacity: '0',
        transform: 'scale(0.7)',
        visibility: 'hidden'
    }
};

//=========================
// EXPORTS
//=========================
export default styles;