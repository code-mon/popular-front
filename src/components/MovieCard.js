import React, { Component } from 'react';
import { getCompleteMovieBackdropPath } from '../utils/api.js';
import { zoomIn, fadeInUp, merge } from 'react-animations';
import Radium from 'radium';
import PropTypes from 'prop-types';
const Waypoint = require('react-waypoint');

const zoomInAndFadeUp = merge(zoomIn, fadeInUp);

class MovieCard extends Component {
  constructor(props) {
    super(props);
    this.styles = {
      base: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        textAlign: 'center',
        width: '300px',
        height: '169px',
        backgroundImage: `url(${getCompleteMovieBackdropPath(
          this.props.movieBackdrop
        )})`,
        fontSize: '18px',
        color: '#f5f5f5',
        opacity: '0.6',
        transition: 'transform 300ms ease-in',
        // animation: `0.5s ease-in`,
        // animationName: Radium.keyframes(zoomInAndFadeUp, 'zoomAndFade'),
        ':hover': {
          zIndex: '1',
          opacity: '1',
          fontSize: '22px',
          transform: 'scale(1.2)'
        }
      },
      hidden: {
        opacity: '0',
        transform: 'scale(0.7)',
        visibility: 'hidden'
      }
    };
    this.state = {
      visible: false
    };
    this.setVisible = this.setVisible.bind(this);
  }

  static propTypes = {
    animationDelay: PropTypes.string,
    movieBackdrop: PropTypes.string.isRequired,
    movieTitle: PropTypes.string.isRequired
  };

  setVisible() {
    this.setState({
      visible: !this.state.visible
    });
  }

  render() {
    return (
      <Waypoint onEnter={this.setVisible} onLeave={this.setVisible}>
        <div
          style={[this.styles.base, !this.state.visible && this.styles.hidden]}>
          {this.props.movieTitle}
        </div>
      </Waypoint>
    );
  }
}

export default Radium(MovieCard);
