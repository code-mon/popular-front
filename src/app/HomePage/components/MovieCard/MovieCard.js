import React, { Component } from 'react'
import { getCompleteMovieBackdropPath } from '../../../utils/api.js'
import MovieTitle from './MovieTitle.js'
import Radium from 'radium'
import PropTypes from 'prop-types'
const Waypoint = require('react-waypoint')

//STYLES
import localStyle from './MovieCard.style';

//COMPONENTS
import FavoriteButton from './FavoriteButton';

class MovieCard extends Component {
    constructor(props) {
        super(props)
        this.state = {
            visible: false,
            isFavorite: false,
        }
        this.setVisible = this.setVisible.bind(this)
        this.setFavorite = this.setFavorite.bind(this)        
    }

    static propTypes = {
        movieBackdrop: PropTypes.string.isRequired,
        movieTitle: PropTypes.string.isRequired
    }

    //get the movie backdrop from the prop for the css background
    getMovieBackgroud() {
        return `url(${getCompleteMovieBackdropPath(
        this.props.movieBackdrop
        )})`;
    }

    setFavorite() {
        this.setState({
            isFavorite: !this.state.isFavorite 
        });
    }

    setVisible() {
        this.setState({
            visible: !this.state.visible
        })
    }

    render() {
        const { movieTitle } = this.props
        //to give a border around selected movies (not 100% sold on the look of this)
        let tempStyle = null;

        //set style based on favorite state
        this.state.isFavorite ? tempStyle = { ...localStyle.base, border: '2px solid #FD4034', backgroundImage: this.getMovieBackgroud(), }
        : tempStyle = { ...localStyle.base, backgroundImage: this.getMovieBackgroud() };
        
        return (
            // Waypoint watches for when an element is in view and calls the onEnter and onLeave as needed
            <Waypoint onEnter={this.setVisible} onLeave={this.setVisible}>
                <div
                    style={[
                        tempStyle,
                        !this.state.visible && localStyle.hidden
                    ]}
                    onClick={ this.setFavorite }>
                    <MovieTitle>{movieTitle}</MovieTitle>
                    <FavoriteButton isToggled={ this.state.isFavorite }/>
                </div>
            </Waypoint>
        )
    }
}

export default Radium(MovieCard)
