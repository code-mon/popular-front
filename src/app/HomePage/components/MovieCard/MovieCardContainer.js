import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { setUserMovies } from '../../actions'
import MovieCard from './MovieCard'

const mapStateToProps = (state, ownProps) => {
    return {
        isFavorite: isFavorite(
            state.dashboard.dashboardUser.movies,
            ownProps.movieTitle
        ),
        userId: state.dashboard.dashboardUser.id
    }
}

const mapDispatchToProps = dispatch => {
    return bindActionCreators({ favoriteMovie: setUserMovies }, dispatch)
}

const isFavorite = (favoriteMovies, movieTitle) => {
    return favoriteMovies.indexOf(movieTitle) > -1 ? true : false
}

export default connect(mapStateToProps, mapDispatchToProps)(MovieCard)
