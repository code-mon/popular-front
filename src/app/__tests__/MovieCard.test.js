import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from '../HomePage/components/MovieCard/MovieCard.js';
const Waypoint = require('react-waypoint');

it('renders correctly', () => {
  const tree = renderer
    .create(<MovieCard movieTitle="Die Hard" movieBackdrop="/" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('has a WayPoint wrapper', () => {
  const wrapper = shallow(
    <MovieCard movieTitle="Die Hard" movieBackdrop="/" />
  );
  expect(wrapper.find(Waypoint)).toHaveLength(1);
});

it('has a movie title', () => {
  const wrapper = shallow(
    <MovieCard movieTitle="Die Hard" movieBackdrop="/" />
  );
  const card = wrapper.children();
  expect(card.text()).toEqual('Die Hard');
});

it('toggles visibility', () => {
  const wrapper = shallow(
    <MovieCard movieTitle="Die Hard" movieBackdrop="/" />
  );
  expect(wrapper.state('visible')).toBe(false);
  wrapper.simulate('enter');
  expect(wrapper.state('visible')).toBe(true);
  wrapper.simulate('leave');
  expect(wrapper.state('visible')).toBe(false);
});
