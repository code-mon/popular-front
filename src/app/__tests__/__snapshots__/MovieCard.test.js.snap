// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-radium={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/w300/)",
      "color": "#f5f5f5",
      "display": "flex",
      "flexDirection": "column",
      "fontSize": "18px",
      "height": "169px",
      "justifyContent": "center",
      "opacity": "0",
      "textAlign": "center",
      "transform": "scale(0.7)",
      "transition": "transform 300ms ease-in",
      "visibility": "hidden",
      "width": "300px",
    }
  }
>
  Die Hard
</div>
`;
